src/MCMCMonitorDataManager/MCMCMonitorData.ts

```diff
@@ -145,13 +145,13 @@ export const mcmcMonitorReducer = (s: MCMCMonitorData, a: MCMCMonitorAction): MC
     else if (a.type === 'setSelectedVariableNames') {
         return {
             ...s,
-            selectedVariableNames: a.variableNames
+            selectedVariableNames: a.variableNames.sort()
```

Comment:
In the spirit of immutability it's probably better to do:
```
selectedVariableNames: [...a.variableNames].sort()
```
This avoids having a reference to this part of the state elsewhere in the code.

Reply:
I'm happy to make the change, but wouldn't the unsorted assignment (i.e. what's in `main` right now) be equally at risk of capturing the reference?

What I'd expect to happen (in both cases) is the state is replaced with a new state whose `selectedVariableNames` key points to the list on the incoming `MCMCMonitorAction`, the rest of `a` goes out of scope and gets GC'd, and the new state object just owns the pointer now. Is there something I'm overlooking?

Reply:
Yes, the same problem existed before your modifications. I think it's better to make a copy.